plugins/wazuh-core/server/services/initialization/initialization.ts

```diff
@@ -0,0 +1,171 @@
+    if (name) {
+      return this.items.get(name);
+    }
+    return Array.from(this.items.values());
+  }
+  createRunContext(scope: InitializationTaskContext, context: any = {}) {
+    return { ...this.services, ...context, scope };
+  }
+  async runAsInternal(taskNames?: string[]) {
+    const ctx = this.createRunContext('internal', { core: this._coreStart });
+    return await this.run(ctx, taskNames);
+  }
+  createNewTaskFromRegisteredTask(name: string) {
+    const task = this.get(name) as InitializationTask;
+    if (!task) {
+      throw new Error(`Task [${name}] is not registered`);
+    }
+    return new InitializationTask({ name, run: task._run });
+  }
+  private async run(ctx, taskNames?: string[]) {
+    try {
+      if (this.items.size) {
+        const allTasks = Array.from(this.items.values());
+        const tasks = taskNames
+          ? allTasks.filter(({ name }) =>
+              taskNames.some(taskName => taskName === name),
+            )
+          : allTasks;
+        const results = await Promise.all(
+          tasks.map(async item => {
+            const logger = this.logger.get(item.name);
+
+            try {
+              return await item.run({
+                ...this.services,
+                ...ctx,
+                logger,
+              });
+            } catch (e) {
+              logger.error(`Error running task [${item.name}]: ${e.message}`);
+              return item.getInfo();
+            }
+          }),
+        );
+        return results;
+      } else {
+        this.logger.info('No tasks');
+      }
+    } catch (error) {
+      this.logger.error(`Error starting: ${error.message}`);
+    }
+  }
+}
+
+class InitializationTask implements IInitializationTask {
+  public name: string;
+  private _run: any;
+  public status: InitializationTaskRunData['status'] = 'not_started';
+  public result: InitializationTaskRunData['result'] = null;
+  public data: any = null;
+  public createdAt: InitializationTaskRunData['createdAt'] =
+    new Date().toISOString();
+  public startedAt: InitializationTaskRunData['startedAt'] = null;
+  public finishedAt: InitializationTaskRunData['finishedAt'] = null;
+  public duration: InitializationTaskRunData['duration'] = null;
+  public error = null;
+  constructor(task: InitializationTaskDefinition) {
+    this.name = task.name;
+    this._run = task.run;
+  }
+  private init() {
+    this.status = 'running';
+    this.result = null;
+    this.data = null;
+    this.startedAt = new Date().toISOString();
+    this.finishedAt = null;
+    this.duration = null;
+    this.error = null;
+  }
+  async run(...params) {
+    if (this.status === 'running') {
+      throw new Error(`Another instance of task ${this.name} is running`);
+    }
+    let error;
+    try {
+      this.init();
+      this.data = await this._run(...params);
+      this.result = 'success';
+    } catch (e) {
+      error = e;
+      this.result = 'fail';
```

Comment:
You could use a constant instead of a hardcoded value.

```suggestion
      this.status = 'fail';
```

Reply:
The `result` prop is related to (`sucess/fail`) and `status` is related to running task (`not_started`,`running`/`finished`).